packages/mco/components/data-policies/create-dr-policy/create-dr-policy.tsx

```diff
@@ -22,32 +33,111 @@ import {
   Dropdown,
   DropdownItem,
   DropdownToggle,
+  Alert,
+  Checkbox
 } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
 import {
   MAX_ALLOWED_CLUSTERS,
   REPLICATION_TYPE,
+  STORAGE_SYSTEM_NAME,
+  ODF_MINIMUM_SUPPORT,
+  CEPH_CLUSTER_NAME,
+  S3_PROFILE_PREFIX
 } from '../../../constants/dr-policy';
-import { DRPolicyModel } from '../../../models/models';
-import { DRPolicyKind } from '../../../types/types';
-import { Cluster, SelectClusterList } from './select-cluster-list';
+import { DRPolicyModel, MirrorPeerModel } from '../../../models/models';
+import { DRPolicyKind, MirrorPeerKind, } from '../../../types/types';
+import { isMinimumSupportedODFVersion } from '../../../utils/disaster-recovery';
+import { Cluster, SelectClusterList, ManagedClusterMapping } from './select-cluster-list';
 import './create-dr-policy.scss';
 
+
+const fetchMirrorPeer = (mirrorPeers: MirrorPeerKind[], peerNames: string[]) => (
+  mirrorPeers?.find((mirrorPeer) => {
+    const existingPeerNames = mirrorPeer?.spec?.items?.map((item) => item?.clusterName) ?? [];
+    if(existingPeerNames.length > 0) {
+      return existingPeerNames.sort().join(',')=== peerNames.sort().join(',')
+    }
+    return false;
+  })
+);
+
+type SelectedClusterProps = {
+  id: number;
+  cluster: Cluster;
+  setSelectedClusters: React.Dispatch<React.SetStateAction<ManagedClusterMapping>>;
+};
+
 const SelectedCluster: React.FC<SelectedClusterProps> = ({
   id,
   cluster,
-  region,
-  storageSystem,
+  setSelectedClusters
 }) => {
+  const { name: clusterName, region, storageSystem, odfVersion} = cluster ?? {};
+  const [ss, ssLoaded, ssLoadError] = useK8sGet<StorageSystemKind>(
+    ODFStorageSystem,
+    STORAGE_SYSTEM_NAME,
+    CEPH_STORAGE_NAMESPACE,
+    clusterName
+  );
+  const [cephCluster, cephClusterLoaded, cephClusterLoadError] = useK8sGet<K8sResourceKind>(
+    CephClusterModel,
+    CEPH_CLUSTER_NAME,
+    CEPH_STORAGE_NAMESPACE,
+    clusterName
+  );
+ 
+  const [csvList, csvListLoaded, csvListLoadError] = useK8sGet<ListKind<ClusterServiceVersionKind>>(
+    ClusterServiceVersionModel,
+    null,
+    CEPH_STORAGE_NAMESPACE,
+    clusterName
+  );
+
+  React.useEffect(() => {
+    if(ssLoaded && cephClusterLoaded && csvListLoaded && !(ssLoadError && cephClusterLoadError && csvListLoadError)){
+      const operator = getODFCsv(csvList?.items);
+      setSelectedClusters((selectedClusters) => ({
+        ...selectedClusters,
+        [clusterName]: {
+          name: clusterName,
+          region: region,
+          storageClusterId: cephCluster?.status?.ceph?.fsid ?? "",
```

Comment:
```suggestion
          storageClusterId: cephCluster?.status?.ceph?.fsid ?? "",
```
Do we need optional chaining if the resources is loaded and no error is present?

Reply:
fsid is introduced on rook recently, only ODF 4.11 will have this value. Usecase is, if this value is not present or unable to read from the managedclusters then we need to ask the user to input the replication type manually. So user wont blocked on DRPolicy creation operation due to this value.